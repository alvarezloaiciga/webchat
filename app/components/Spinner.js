// @flow
import React from 'react';
import {messageTypes} from 'Common/Constants';
import quiqOptions, {getMessage} from 'utils/quiq';
import {isIE9} from 'Common/Utils';
import './styles/Spinner.scss';

const {color} = quiqOptions;

const Spinner = () =>
  <div className="Spinner">
    {!isIE9()
      ? <div className="loading" style={{borderColor: color}} />
      : <span className="plainText">
          {getMessage(messageTypes.connectingMessage)}
        </span>}
  </div>;

export default Spinner;
