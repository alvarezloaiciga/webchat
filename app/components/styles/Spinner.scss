@import '~styles/animations';

.Spinner {
  .loading {
    animation: spin 1250ms infinite linear;
    border: 8px solid #59ad5d;
    border-right-color: transparent !important;
    border-radius: 32px;
    box-sizing: border-box;
    display: inline-block;
    position: relative;
    overflow: hidden;
    width: 64px;
    height: 64px;
  }
}

.noFlexbox .Spinner {
  display: table;
  height: 520px;
  text-align: center;
  width: 100%;

  .plainText {
    display: table-cell;
    vertical-align: middle;
  }
}
