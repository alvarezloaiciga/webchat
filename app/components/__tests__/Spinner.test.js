// @flow

jest.mock('Common/QuiqOptions');

import React from 'react';
import Spinner from '../Spinner';
import {shallow} from 'enzyme';
import type {ShallowWrapper} from 'enzyme';

describe('Spinner component', () => {
  let wrapper: ShallowWrapper;
  let render: () => void;

  beforeEach(() => {
    render = () => {
      wrapper = shallow(<Spinner />);
    };
  });

  describe('rendering', () => {
    beforeEach(() => {
      render();
    });

    it('renders', () => {
      expect(wrapper).toMatchSnapshot();
    });
  });
});
