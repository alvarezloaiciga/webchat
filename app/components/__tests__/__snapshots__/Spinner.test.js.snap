// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Spinner component rendering renders 1`] = `
<div
  className="Spinner"
>
  <div
    className="loading"
    style={
      Object {
        "borderColor": "#333",
      }
    }
  />
</div>
`;
